### PR TITLE
ADD Conv2DMPO Layer

### PR DESCRIPTION
**Pull request by jacksonwb**
* Change to use tf.nn.conv2d
* Add get_config
* Add compute_output_shape
* Add tuple normalization for strides, dilation and kernel_size
* Add support for padding and data_format
* Add docstring
* Add unit tests
* Change data_format to use Keras form

### CODE REVIEW

**Comment by Thenerdstation** [comment]
Awesome! Thanks Jackson!

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+      self.use_bias = None
+
+  def call(self, inputs: tf.Tensor, **kwargs) -> tf.Tensor:
+
+    tn_nodes = [tn.Node(n, backend='tensorflow') for n in self.nodes]
+    for i in range(len(tn_nodes) - 1):
+      tn_nodes[i][2] ^ tn_nodes[i+1][1]
+    input_edges = [n[0] for n in tn_nodes]
+    output_edges = [n[3] for n in tn_nodes]
+    edges = [tn_nodes[0][1], tn_nodes[-1][2]] + \
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:202]: I think you can make this just one line.

In general, we prefer to use `()` instead of `\` to break a line

So do 
```
a = (b +
     c)
```
instead of 

```
a = b + \
    c
```

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+                       'should be defined. Found `None`.')
+
+    def is_perfect_root(n, n_nodes):
+      root = n**(1. / n_nodes)
+      return round(root)**n_nodes == n
+
+    channels = input_shape[channel_axis]
+
+    # Ensure dividable dimensions
+    assert is_perfect_root(channels, self.num_nodes), \
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:149]: Remove all of the `\`s

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    tn_nodes = [tn.Node(n, backend='tensorflow') for n in self.nodes]
+    for i in range(len(tn_nodes) - 1):
+      tn_nodes[i][2] ^ tn_nodes[i+1][1]
+    input_edges = [n[0] for n in tn_nodes]
+    output_edges = [n[3] for n in tn_nodes]
+    edges = [tn_nodes[0][1], tn_nodes[-1][2]] + \
+        input_edges + output_edges
+
+    contract = tn.contractors.greedy(tn_nodes, edges)
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:205]: nit: Can we renamed this to `contracted`? Make it a noun instead of a verb.

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+    if self.use_bias:
+      bias = tf.reshape(self.bias_var, (1, self.filters,))
+      result += bias
+
+    if self.activation is not None:
+      result = self.activation(result)
+    return result
+
+  def compute_output_shape(self, input_shape: List[int]) -> \
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:225]: remove `\`

File: tensornetwork/tn_keras/test_conv_layer.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    if os.path.isdir(save_path):
+      shutil.rmtree(save_path)
+
+    # Clean up h5 file
+    if os.path.exists(save_path):
+      os.remove(save_path)
+
+    # Compare model predictions and loaded_model predictions
+    np.testing.assert_almost_equal(model.predict(data),
+                                   loaded_model.predict(data))
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/test_conv_layer.py:146]: Be sure to use the tmp_path argument from pytest so that files don't accidentally overwrite each other when the tests run in parallel.

https://docs.pytest.org/en/latest/tmpdir.html

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    kernel_initializer: Initializer for the node weight matrices.
+    bias_initializer: Initializer for the bias vector.
+    kernel_regularizer: Regularizer for the node weight matrices.
+    bias_regularizer: Regularizer for the bias vector.
+
+  Input shape:
+    4D tensor with shape: `(batch_size, h, w, channels)`.
+
+  Output shape:
+    2D tensor with shape: `(batch_size, h, w, filters)`.
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:85]: 4D

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+import tensornetwork as tn
+import math
+
+
+@tf.keras.utils.register_keras_serializable()
+class Conv2DMPO(Layer):
+  """2D Convolutional Matrix Product Operator (MPO) TN layer.
+
+  This layer creates an MPO kernel which is contracted and convolved
+  with the layer input to produce a tensor of outputs.
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:17]: Not a huge fan of this description.

File: tensornetwork/tn_keras/conv2d_mpo.py [hunk 8]
```diff
@@ -13,8 +13,12 @@
...
   """2D Convolutional Matrix Product Operator (MPO) TN layer.
 
-  This layer creates an MPO kernel which is contracted and convolved
-  with the layer input to produce a tensor of outputs.
+  This layer recreates the functionality of a traditional convolutional
+  layer, but stores the 'kernel' as a network of nodes forming an MPO.
+  The bond dimension of the MPO can be adjusted to increase or decrease the
+  number of parameters independently of the input and output dimensions.
+  When the layer is called, the MPO is contracted into a traditional kernel
+  and convolved with the layer input to produce a tensor of outputs.
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/conv2d_mpo.py:21]: This is much better. Good work!
